widget/index.html

```diff
@@ -2,66 +2,218 @@
+			</div>
+			<hr role="separator" class="mdc-list-divider">
+			<div class="mdc-layout-grid__cell">
+				<label for="min-temp--information">Min temperature</label>
+				<span class="mdc-layout-grid--align-right" id="min-temp--information"></span>
+			</div>
+			<hr role="separator" class="mdc-list-divider">
+			<div class="mdc-layout-grid__cell">
+				<label for="max-temp--information">Max temperature</label>
+				<span class="mdc-layout-grid--align-right" id="max-temp--information"></span>
+            </div>
+            <hr role="separator" class="mdc-list-divider">   
+        </div>
+        <div class="refresh--button">
+            <button class="mdc-button" onclick="handleButtonRefresh()">
+                <div class="mdc-button__ripple"></div>
+                <span class="mdc-button__label">Refresh</span>
+              </button>
+        </div>
 	</div>
-	<div class="alert alert-success" role="alert">
-		<h1 id="clouds"></h1>
-	</div>
-	<script>
-
-		function loadWeatherData(city) {
-			city = encodeURIComponent(city);
-			/// Call open weather API to pull data
-			$.ajax({
-				url: "http://api.openweathermap.org/data/2.5/find?appid=2de143494c0b295cca9337e1e96b00e0&"
-				+ "q=" + city + "&units=imperial"
-			}).done(function (data) {
-				/// check you have data to report on
-				if (data && data.list && data.list.length) {
-					/// fill in the UI with the new data
-					var report = data.list[0];
-					$('#location').text(report.name);
-					$('#summary').text(report.weather.description);
-					$('#temp').text(report.main.temp + " degrees");
-					$('#rang').text("Low: " + report.main.temp_min + " - High: " + report.main.temp_max);
-					$('#clouds').text(report.clouds.all > 50 ? "Cloudy" : "Clear Skys");
-				}
-			});
-		}
-
-		buildfire.datastore.get(function(err,obj){
-			if(obj)
-				loadWeatherData(obj.data.loc);
-		});
-
-		buildfire.datastore.onUpdate(function(obj){
-			loadWeatherData(obj.data.loc);
-		});
-
-	</script>
+<script>
+window.onload = (event) => {
+    Settings.get((err, data) => {
+        if(err) return console.log(err);
+        renderPage(data);
+    });
+
+    Settings.update((event) => {
+        refreshWeatherData(event.data);
+    });
+
+    fetchWeatherIcons();
+}
+
+handleButtonRefresh = () => {
+    Settings.get((err, data) => {
+        if(err) return console.log(err);
+        refreshWeatherData(data);
+    });
+}
+
+fetchWeatherIcons = (iconCode) => {
+    fetch("https://openweathermap.org/img/wn/" + iconCode + "@2x.png"
+    ).then(function (response) {
+        createWeatherImage(response.url);
+    }).catch(function () {
+        console.log('failed to load image');
+    });
+}
+
+fetchWeatherData = (latitude, longitude, tempUnit, apiKey, callback) => {
+    /// Call open weather API to pull data
+    fetch("https://api.openweathermap.org/data/2.5/find?appid=" + apiKey + "&lat=" + latitude + "&lon=" + longitude + tempUnit
```

Comment:
lets move the endpoints to a higher level variable so that any change in the endpoint wont be "painful" on our side